src/test/scala/com/amazon/deequ/analyzers/AnalyzerTests.scala

```diff
@@ -495,6 +498,108 @@ class AnalyzerTests extends AnyWordSpec with Matchers with SparkContextSpec with
     }
   }
 
+  "DateTimeDistribution analyzer" should {
+
+    def distributionFrom(
+      nonZeroValues: (Instant, Instant, DistributionValue)*)
+    : Distribution = {
+
+      val distributionValues = nonZeroValues
+        .map { case (from, to, distValue) => s"($from to $to)" -> distValue }.toMap
+
+      Distribution(distributionValues, numberOfBins = distributionValues.keys.size)
+    }
+
+    "fail for non DateType or TimestampType column" in withSparkSessionJava8APIEnabled { sparkSession =>
+      val df = getDfFull(sparkSession)
+      assert(DateTimeDistribution("item", DistributionInterval.HOURLY).calculate(df).value.isFailure)
+    }
+
+    "success for DateType column" in withSparkSessionJava8APIEnabled { sparkSession =>
+      val df = getDfWithLocalDateAndInstant(sparkSession)
+      assert(DateTimeDistribution("signupDate", DistributionInterval.MONTHLY).calculate(df).value.isSuccess)
+    }
+
+    "success for Timestamp column" in withSparkSessionJava8APIEnabled { sparkSession =>
+      val df = getDfWithLocalDateAndInstant(sparkSession)
+      assert(DateTimeDistribution("dateOfBirth", DistributionInterval.HOURLY).calculate(df).value.isSuccess)
+    }
+
+    "success get datetimeDistribution with Daily Interval" in withSparkSessionJava8APIEnabled { sparkSession =>
+      val df = getDfWithLocalDateAndInstant(sparkSession)
+      val actualDistribution = DateTimeDistribution("dateOfBirth", DistributionInterval.DAILY).calculate(df).value
+      actualDistribution shouldBe Success(
+        distributionFrom(
+          (Instant.parse("2021-11-11T00:00:00Z"), Instant.parse("2021-11-11T23:59:59.999Z"), DistributionValue(3, 0.6)),
+          (Instant.parse("2019-04-11T00:00:00Z"), Instant.parse("2019-04-11T23:59:59.999Z"), DistributionValue(2, 0.4))
+        )
+      )
+    }
+
+    "datetimeDistribution analyzer with VerificationSuite" in withSparkSessionJava8APIEnabled { sparkSession =>
```

Comment:
@eycho-am Hi just to confirm this is the kind of test you was looking for right?
Currently It is only implemented as an Analyzer to run with AnalysisRunner or as RequiredAnalyzer

For more advance use case with `VerificationSuite`, I can open a PR to implement some check based on it